### PR TITLE
 Fix missing validation in ListViewGroupCollection.Insert

### LINKED ISSUE #372: ListViewGroupCollection.Insert is missing validation present in Add
Add:
```cs
public int Add(ListViewGroup group)
{
    if (group == null)
    {
        throw new ArgumentNullException(nameof(group));
    }

    if (this.Contains(group)) {
        return -1;
    }

    // Will throw InvalidOperationException if group contains items which are parented by another listView.
    CheckListViewItems(group);
    group.ListViewInternal = this.listView;
    int index = this.List.Add(group);
    if (listView.IsHandleCreated) {
        listView.InsertGroupInListView(this.List.Count, group);
        MoveGroupItems(group);
    }            
    return index;
}        
```

Insert:
```cs
public void Insert(int index, ListViewGroup group) {
    if (group == null)
    {
        throw new ArgumentNullException(nameof(group));
    }

    if (Contains(group)) {
        return;
    }
    group.ListViewInternal = this.listView;
    this.List.Insert(index, group);
    if (listView.IsHandleCreated) {
        listView.InsertGroupInListView(index, group);
        MoveGroupItems(group);
    }            
}       
```

Notice the missing call in `Insert` to `CheckListViewItems(group);`

This lets us get into an invalid state in `Insert`

### PR DESCRIPTION
**Pull request by hughbe**
Fixes #372

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/ListViewGroupCollection.cs [hunk 1]
```diff
@@ -259,6 +258,7 @@ public void Insert(int index, ListViewGroup group)
                 return;
             }
 
+            CheckListViewItems(group);
```
Comment (hughbe) [inline src/System.Windows.Forms/src/System/Windows/Forms/ListViewGroupCollection.cs]: Now matched logic in `.Add`

File: src/System.Windows.Forms/src/System/Windows/Forms/ListViewGroupCollection.cs [hunk 2]
```diff
@@ -128,7 +128,6 @@ public int Add(ListViewGroup group)
                 return -1;
             }
 
-            // Will throw InvalidOperationException if group contains items which are parented by another listView.
```
Comment (hughbe) [inline src/System.Windows.Forms/src/System/Windows/Forms/ListViewGroupCollection.cs]: Comment was wrong, actually throws `ArgumentException`. Doesn't add weight anyway, so removed

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ListViewGroupCollectionTests.cs [hunk 3]
```diff
@@ -821,21 +835,19 @@ public void ListViewGroupCollection_Insert_SetAlreadyInCollection_Nop(int index)
...
 
             var item = new ListViewItem();
             var group = new ListViewGroup();
             group.Items.Add(item);
             otherListView.Items.Add(item);
 
-            collection.Insert(0, group);
-            Assert.Same(group, collection[0]);
-            Assert.Same(listView, group.ListView);
+            Assert.Throws<ArgumentException>(null, () => collection.Insert(0, group));
```
Comment (hughbe) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ListViewGroupCollectionTests.cs]: This is the only test that actually changed. The others just confirmed the return value of `.Add`

**Comment by zsd4yr** [comment]
What validation(s) were missing?

**Comment by hughbe** [comment]
Validation is present in Add that wasn’t previously in Insert, namely checking for duplicates

**Comment by zsd4yr** [comment]
Can you add a test that attempts to add a duplicate and ensures it doesn't get added ?

**Comment by hughbe** [comment]
We have the following tests:
- ListViewGroupCollection_Insert_AlreadyInOtherCollection_GetReturnsExpected
- ListViewGroupCollection_Insert_SetAlreadyInCollection_Nop
- ListViewGroupCollection_Insert_HasItemsFromOtherListView_ThrowsArgumentException

The last test is the one that has been updated to reflect the new behavior that matches `.Add`

**Comment by danmosemsft** [comment]
@zsd4yr is your feedback addressed ?
